packages/core-v2/src/semantic-model/concepts/concepts-utils.ts

```diff
@@ -14,4 +19,52 @@ export function isSemanticModelRelationship(resource: Entity | null): resource i
 export const SEMANTIC_MODEL_GENERALIZATION = "generalization"; // todo use proper IRI
 export function isSemanticModelGeneralization(resource: Entity | null): resource is SemanticModelGeneralization {
     return resource?.type.includes(SEMANTIC_MODEL_GENERALIZATION) ?? false;
-}
\ No newline at end of file
+}
+
+export function isSemanticModelAttribute(resource: Entity | null): resource is SemanticModelRelationship {
+    if (!isSemanticModelRelationship(resource)) {
+        return false;
+    }
+    const [end1, end2] = resource.ends;
+
+    return (
+        (end1?.iri != null &&
+            end2?.iri == null &&
+            semanticModelRelationshipEndConceptIsNullEmptyOrDataType(end1, resource)) ||
+        (end2?.iri != null &&
+            end1?.iri == null &&
+            semanticModelRelationshipEndConceptIsNullEmptyOrDataType(end2, resource))
```

Comment:
We discussed this with @martinnec. The question "Is this relation an attribute or association?" depends on the context you are in.

In your editor, you need to know whether the range should be represented as a separate box connected with a line or whether it should be in the box itself. Although there could be some recommendations, there should eventually be an **option to change it to the other representation**. Imagine an Address class. It has a street name, city, and zip code. Now, imagine that you have a Person with a home address. It is an association, right? But what if you are in a diagram where it is not important to know the structure of the address class, and the opposite direction of association is not important as well? In this context, you would like to see it inside the box, not as a separate box.

In the structural editor, the answer is whether we know how to convert the concept to the representation in the required format. Primitive data types are easy. But the Address can be easy as well. If you already have an existing JSON Schema for an Address, then the Address becomes primitive in your schema as it is effectively just a reference to that schema file.

In OWL, we need to distinguish between `owl:ObjectProperty` and `owl:DatatypeProperty`. One is used for individuals, the other for primitive values, so it should depend only on the concept, whether is primitive or not?

---

So I would suggest naming this function `isSemanticModelRelationPrimitive` and check if
- resource.iri is null
- resource.name and resource.description are empty objects
- one end has:
  - end.iri is null
  - end.name and end.description are empty objects
- the other end has:
  - concept as a primitive type or is unknown (=null)
  - (no requirements on end.iri)  

You can use this function to decide whether to render things as attributes or associations. Also, it would be interesting to have an option in a visual model to render a relation as an attribute.

Reply:
Updated. Having such option might be nice, you can create an issue for it 😉